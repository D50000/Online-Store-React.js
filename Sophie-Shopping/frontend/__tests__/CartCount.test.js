import { render } from '@testing-library/react';
import wait from 'waait';

import CartCount from '../components/CartCount';

describe('<CartCount/>', () => {
  it('Renders', () => {
    // Arrange
    render(<CartCount count={10} />);
  });

  it('Matches snapshot', () => {
    // Arrange
    const { container } = render(<CartCount count={11} />);
    expect(container).toMatchSnapshot();
  });

  it('updates via props', async () => {
    // Act
    const { container, rerender, debug } = render(<CartCount count={11} />);
    // Assert
    expect(container.textContent).toBe('11');
    expect(container).toHaveTextContent('11');

    // Act, update the props
    rerender(<CartCount count={12} />);
    // Wait for the animation finish when add count.
    await wait(400);
    // Assert
    expect(container).toHaveTextContent('12');
    debug();
    // Save update props screenshot
    expect(container).toMatchSnapshot();
  });
});
