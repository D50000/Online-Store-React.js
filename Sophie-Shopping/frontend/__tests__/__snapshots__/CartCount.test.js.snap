// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CartCount/> Matches snapshot 1`] = `
<div>
  <span
    class="CartCount__AnimationStyles-sc-r79obc-1 hzcddk"
  >
    <div>
      <div
        class="CartCount__Dot-sc-r79obc-0 hAXSqz count"
      >
        11
      </div>
    </div>
  </span>
</div>
`;

exports[`<CartCount/> updates via props 1`] = `
<div>
  <span
    class="CartCount__AnimationStyles-sc-r79obc-1 hzcddk"
  >
    <div>
      <div
        class="CartCount__Dot-sc-r79obc-0 hAXSqz count count-enter-done"
      >
        12
      </div>
    </div>
  </span>
</div>
`;
